### ISSUE TITLE
Unexpected results with user sort and toggling hide options

### ISSUE THREAD

**Issue by bsnouffer** [shcaba/FishPath-updates#96]
The user sort order does not work correctly when there are hidden options and the user toggles 'show hide options'. 

Step 1: sort and hide options on same screen
![before hiding](https://user-images.githubusercontent.com/11687620/41252705-d35fef54-6d72-11e8-9f11-463db0dba376.png)

Step 2: Toggle 'show hidden options'
![hide](https://user-images.githubusercontent.com/11687620/41252766-f9ad294c-6d72-11e8-8b5a-44fafe9e7b4d.png)

Step 3: Sort another option
![sort](https://user-images.githubusercontent.com/11687620/41252779-0c35f31e-6d73-11e8-8300-e83e5031cf9d.png)

Step 4: Toggle 'show hidden options' again. Here you will see that all hidden options will appear at the top. 
![unhide](https://user-images.githubusercontent.com/11687620/41252874-573dcd5a-6d73-11e8-8b0d-397a44b5e705.png)

Note that variations of this appear in similar workflows. It will probably most commonly manifest itself when a user first hides options, then toggles 'show hidden' so they disappear from the screen. Once they are done sorting, if they toggle "show hidden" again, the hidden options will appear at the top of the list. 

From a dev standpoint, this is due to the data structure storing the user sorting order being designed under the paradigm that hidden options were displayed separately in their own 'tab'. The data structure needs to be updated to match the new paradigm of hidden options remaining in the list.

**Comment by bsnouffer** [comment]
I am still seeing this as an issue again. I know that a fix was done in Jan, but something has affected sorting in general, not just this specific workflow. I will open a new related issue so we can track both as I am not sure yet if they are caused by the same issue and I want to highlight the general sorting issue.

**Comment by underbluewaters** [comment]
I think the issue on the other repo was closed because we decided it wasn't worth fixing since the workflow should normally be Review Full Results -> Option Retention (hiding) -> Start Options -> Then finally rank the top options. If the user proceeds through this winnowing down of options it shouldn't be an issue.

To make this work in all scenarios would possibly require a change in the way the list is rendered. Whenever the user sorts the list the sortIndexes are all updated based on what is on the screen.  I think we'd need to render some sort of invisible placeholder for all the hidden rows in order for that sortIndex to save as expected.

**Comment by bsnouffer** [comment]
I discussed this workflow a month or two ago with Dawn. We decided that it would be best to show the hidden options at the bottom of the list instead of the top in this situation. I can take ownership of this issue.
I isolated the issue for this problem and have a possible fix, but have not yet fully implemented it.  I have figured out how to move the hidden options to the bottom of the list and am now looking into the possibility of maintaining the user order of the hidden options.
